src/category_theory/glue_data.lean

```diff
@@ -0,0 +1,201 @@
+/-
+Copyright (c) 2021 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import tactic.elementwise
+import category_theory.limits.shapes.multiequalizer
+import category_theory.limits.constructions.epi_mono
+
+/-!
+# Gluing data
+
+We define `glue_data` as a family of data needed to glue topological spaces, schemes, etc. We
+provide the API to realize it as a multispan diagram, and also states lemmas about its
+interaction with a functor that preserves certain pullbacks.
+
+-/
+
+noncomputable theory
+
+open category_theory.limits
+namespace category_theory
+
+universes v u₁ u₂
+
+variables (C : Type u₁) [category.{v} C] {C' : Type u₂} [category.{v} C']
+
+/--
+A family of gluing data consists of
+1. An index type `ι`
+2. An object `U i` for each `i : ι`.
+3. An object `V i j` for each `i j : ι`.
+4. A monomorphism `f i j : V i j ⟶ U i` for each `i j : ι`.
+5. A transition map `t i j : V i j ⟶ V j i` for each `i j : ι`.
+such that
+6. `f i i` is an isomorphism.
+7. `t i i` is the identity.
+8. The pullback for `f i j` and `f i k` exists.
+8. `V i j ×[U i] V i k ⟶ V i j ⟶ V j i` factors through `V j k ×[U j] V j i ⟶ V j i` via some
+    `t' : V i j ×[U i] V i k ⟶ V j k ×[U j] V j i`.
+9. `t' i j k ≫ t' j k i ≫ t' k i j = 𝟙 _`.
+-/
+@[nolint has_inhabited_instance]
+structure glue_data :=
+  (ι : Type v)
+  (U : ι → C)
+  (V : ι × ι → C)
+  (f : Π i j, V (i, j) ⟶ U i)
+  (f_mono : ∀ i j, mono (f i j) . tactic.apply_instance)
+  (f_has_pullback : ∀ i j k, has_pullback (f i j) (f i k) . tactic.apply_instance)
```

Comment:
I would move these to the end. That way the hole-command for structures will also put them at the end when it generates a skeleton.

Reply:
`f_has_pullback` has to come be before `t'`, since the definition of the latter uses pullbacks.
I'm not sure whether the typeclass search will use `f_mono` to resolve `f_has_pullback` if I moved `f_mono` to the end.

Reply:
Ok, good point. Let's leave it like this then.